pkg/sync/expand/graph_test.go

```diff
@@ -103,8 +103,7 @@ func TestGetDescendants(t *testing.T) {
 			graph := parseExpression(t, ctx, testCase.expression)
 
 			descendantEntitlements := graph.GetDescendantEntitlements(testCase.rootID)
-			expectedEntitlements := createExpectedEntitlementsMap(testCase.expectedIDs)
-			require.EqualValues(t, expectedEntitlements, descendantEntitlements, graph.Str())
+			expectEntitlements(t, descendantEntitlements, testCase.expectedIDs)
```

Comment:
This fails intermittently. Go caches test results by default, so you'll need to run `go test -count=1 -v github.com/conductorone/baton-sdk/pkg/sync/expand` to get the flakiness. Around 10% of the time I get this error:

```
=== RUN   TestGetDescendants
=== RUN   TestGetDescendants/no_descendants
=== RUN   TestGetDescendants/N_descendants
    graph_test.go:62: 
        	Error Trace:	/Users/ggreer/code/github.com/ConductorOne/baton-sdk/pkg/sync/expand/graph_test.go:62
        	            				/Users/ggreer/code/github.com/ConductorOne/baton-sdk/pkg/sync/expand/graph_test.go:106
        	Error:      	Not equal: 
        	            	expected: []string{"2", "3"}
        	            	actual  : []string{"3", "2"}
        	            	
        	            	Diff:
        	            	--- Expected
        	            	+++ Actual
        	            	@@ -1,4 +1,4 @@
        	            	 ([]string) (len=2) {
        	            	- (string) (len=1) "2",
        	            	- (string) (len=1) "3"
        	            	+ (string) (len=1) "3",
        	            	+ (string) (len=1) "2"
        	            	 }
        	Test:       	TestGetDescendants/N_descendants
=== RUN   TestGetDescendants/grandchildren_don't_count
=== RUN   TestGetDescendants/unfixed_self-cycle
--- FAIL: TestGetDescendants (0.00s)
    --- PASS: TestGetDescendants/no_descendants (0.00s)
    --- FAIL: TestGetDescendants/N_descendants (0.00s)
    --- PASS: TestGetDescendants/grandchildren_don't_count (0.00s)
    --- PASS: TestGetDescendants/unfixed_self-cycle (0.00s)
=== RUN   TestRemoveNode
--- PASS: TestRemoveNode (0.00s)
=== RUN   TestGetCycles
--- PASS: TestGetCycles (0.00s)
=== RUN   TestHandleCycle
=== RUN   TestHandleCycle/example
=== RUN   TestHandleCycle/simplest
=== RUN   TestHandleCycle/simple
=== RUN   TestHandleCycle/two_cycles
--- PASS: TestHandleCycle (0.00s)
    --- PASS: TestHandleCycle/example (0.00s)
    --- PASS: TestHandleCycle/simplest (0.00s)
    --- PASS: TestHandleCycle/simple (0.00s)
    --- PASS: TestHandleCycle/two_cycles (0.00s)
=== RUN   TestHandleComplexCycle
--- PASS: TestHandleComplexCycle (0.00s)
=== RUN   TestHandleCliqueCycle
--- PASS: TestHandleCliqueCycle (0.00s)
=== RUN   TestMarkEdgeExpanded
--- PASS: TestMarkEdgeExpanded (0.00s)
FAIL
FAIL	github.com/conductorone/baton-sdk/pkg/sync/expand	0.180s
FAIL
```

I'm guessing you just need to change expectEntitlements() to sort the values, then compare.

Reply:
I replaced the `EqualValues()` call with `ElementsMatch()`.